src/database/DataTypes/Requisition.js

```diff
@@ -210,23 +210,26 @@ export class Requisition extends Realm.Object {
   /**
    * Add all items from the mobile store master list to this requisition.
    *
-   * @param  {Realm}  database
-   * @param  {Name}   thisStore
+   * @param  {Realm}            database
+   * @param  {Array.<string>}   selected masterlists from multiselect
+   * @param  {Name}             thisStore
    */
-  addItemsFromMasterList(database, thisStore) {
+  addItemsFromMasterList(database, selected, thisStore) {
```

Comment:
I think this should be altered to just add items from a single `MasterList` that is passed as a parameter. There's no need to have it do all this work now.

Could potentially do it a few different ways, as really we just need a mechanism to either: iterate over a single master list and create a record for each item, or, iterate over a list of items and create a record (for instance if you passed a list of items which are related to the selected master lists). You could even make a case for this logic being in `createRecord`

Reply:
I also did not like that `addItemsFromMasterList` for Requisition had very less to do with Requisition record data itself. I would extract it out of here. It was here and I was trying not to change old code as much as I could. New issue?

Reply:
yeah new issue is fine... I am trying to change any old code as we like to make it better 🤣